<a href="https://colab.research.google.com/github/CppR-Repo/TT-2C2025-Data-Analitycs-Notebooks/blob/main/PreEntrega/MiPre-entrega/Pre-entrega.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Análisis de datos con Pyhon - Pre-entrega
Nombre y apellido: Cristian Pablo Portillo Rios

# Etapa 1

## Script básico
Actividades 2 y 3

In [211]:
# Generamos una lista de diccionarios que contenga algunos datos
ventas = [
  {'producto': 'casablanca', 'tipo': 'latex', 'descripcion': 20, 'precio': 120000.0, 'cantidad': 2},
  {'producto': 'satinol', 'tipo': 'sintetico', 'descripcion': 4, 'precio': 75000.0, 'cantidad': 3},
  {'producto': 'andina', 'tipo': 'fijador', 'descripcion': 1, 'precio': 7800.0, 'cantidad': 2},
  {'producto': 'comodin', 'tipo': 'enduido', 'descripcion': 10, 'precio': 20500.0, 'cantidad': 1},
  {'producto': 'aa', 'tipo': 'lija', 'descripcion': 100, 'precio': 2100.0, 'cantidad': 10}
]

In [212]:
# Calculamos las ventas acumuladas - map
# 1. Calcula la venta de cada producto con map
ventas_individuales = map(lambda x:x['precio'] * x['cantidad'], ventas)

# 2. Sumar todas las ventas individuales con sum
ventas_acumuladas = sum(ventas_individuales)
print(f"Las ventas acumuladas son: {ventas_acumuladas}")

Las ventas acumuladas son: 522100.0


In [213]:
# Calculamos las ventas acumuladas - listas por comprensión
venta_acumuladas = sum([venta["precio"] * venta["cantidad"] for venta in ventas])
venta_acumuladas

522100.0

## Import Datasets
Actividad 1

In [214]:
import pandas as pd

In [215]:
# Import dataset ventas
df_ventas = pd.read_csv("https://raw.githubusercontent.com/CppR-Repo/TT-2C2025-Data-Analitycs-Notebooks/refs/heads/main/PreEntrega/Datasets/ventas.csv")

In [216]:
# Import dataset clientes
df_clientes = pd.read_csv("https://raw.githubusercontent.com/CppR-Repo/TT-2C2025-Data-Analitycs-Notebooks/refs/heads/main/PreEntrega/Datasets/clientes.csv")

In [217]:
# Import dataset marketing
df_marketing = pd.read_csv("https://raw.githubusercontent.com/CppR-Repo/TT-2C2025-Data-Analitycs-Notebooks/refs/heads/main/PreEntrega/Datasets/marketing.csv")

In [218]:
df_ventas

id_venta                producto   precio  cantidad fecha_venta  \
0          792       Cuadro decorativo   $69.94       5.0  02/01/2024   
1          811         Lámpara de mesa  $105.10       5.0  02/01/2024   
2         1156                Secadora   $97.96       3.0  02/01/2024   
3         1372                Heladera  $114.35       8.0  02/01/2024   
4         1546                Secadora  $106.21       4.0  02/01/2024   
...        ...                     ...      ...       ...         ...   
3030      1837         Horno eléctrico  $104.12       9.0  30/12/2024   
3031      2276                  Laptop   $85.27       9.0  30/12/2024   
3032      2696                  Laptop  $107.81       4.0  30/12/2024   
3033      2913              Smartphone   $99.85       7.0  30/12/2024   
3034      2930  Consola de videojuegos   $55.47       6.0  30/12/2024   

              categoria  
0            Decoración  
1            Decoración  
2     Electrodomésticos  
3     Electrodomésticos  
4     Electrodomésticos  
...                 ...  
3030  Electrodomésticos  
3031        Electrónica  
3032        Electrónica  
3033        Electrónica  
3034        Electrónica  

[3035 rows x 6 columns]

## Dataframe ventas

### Análisis exploratorio de **datos**
Actividad 4

Aplicamos los métodos de EDA vistos en clase, `info`, `columns`, `dtypes`, `head`, `tail`, `sample`, `unique`, `value_counts`, `describe`, `duplicated`, `isnull`, etc. y documentamos las observaciones:

In [219]:
# Aplicamos info() para familiarizarnos con el dataframe
df_ventas.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3035 entries, 0 to 3034
Data columns (total 6 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   id_venta     3035 non-null   int64  
 1   producto     3035 non-null   object 
 2   precio       3033 non-null   object 
 3   cantidad     3033 non-null   float64
 4   fecha_venta  3035 non-null   object 
 5   categoria    3035 non-null   object 
dtypes: float64(1), int64(1), object(4)
memory usage: 142.4+ KB


In [220]:
# Visualizamos los primeros registros con head()
# también podemos usar tail() para visualizar los últimos
df_ventas.head()

id_venta           producto   precio  cantidad fecha_venta  \
0       792  Cuadro decorativo   $69.94       5.0  02/01/2024   
1       811    Lámpara de mesa  $105.10       5.0  02/01/2024   
2      1156           Secadora   $97.96       3.0  02/01/2024   
3      1372           Heladera  $114.35       8.0  02/01/2024   
4      1546           Secadora  $106.21       4.0  02/01/2024   

           categoria  
0         Decoración  
1         Decoración  
2  Electrodomésticos  
3  Electrodomésticos  
4  Electrodomésticos

In [221]:
# Usar sample() para tomar registros al azar
df_ventas.sample(5)

id_venta               producto   precio  cantidad fecha_venta  \
217        787  Elementos de cerámica  $106.39       6.0  26/01/2024   
630       1772               Heladera   $56.94       2.0  14/03/2024   
317       2267         Cámara digital   $57.81      10.0  06/02/2024   
2282      2793             Smartphone   $91.37       8.0  26/09/2024   
2125      2405            Auriculares  $109.22       8.0  06/09/2024   

              categoria  
217          Decoración  
630   Electrodomésticos  
317         Electrónica  
2282        Electrónica  
2125        Electrónica

In [222]:
# Analizamos el campo precio
# Si es numericos se puede aplicar describe()
df_ventas["precio"].sample(5)

712     $103.34
1167    $113.70
1545     $27.13
86      $117.43
1924     $66.74
Name: precio, dtype: object

In [223]:
# Analizamos el campo cantidad
# si es numerico se puede aplicar describe()
df_ventas["cantidad"].sample(5)

2869     6.0
207      4.0
2722    12.0
91      10.0
320      9.0
Name: cantidad, dtype: float64

In [224]:
# Analizamos el campo fecha_venta
df_ventas["fecha_venta"].sample(5)

1489    16/06/2024
386     14/02/2024
705     21/03/2024
746     26/03/2024
2558    01/11/2024
Name: fecha_venta, dtype: object

In [225]:
# Analizamos el campo categoria
# Probar con value_counts y unique
df_ventas["categoria"].value_counts()

categoria
Decoración           1015
Electrónica          1012
Electrodomésticos    1008
Name: count, dtype: int64

### Calidad de datos
Actividad 5

In [226]:
# Buscamos registros duplicados
df_ventas.duplicated().sum()

np.int64(35)

In [227]:
# Visualizamos los duplicados
# keep: 'first', 'last', False
df_ventas[df_ventas.duplicated(keep='first')]

id_venta                producto   precio  cantidad fecha_venta  \
826        56                Cortinas   $66.24       5.0  05/04/2024   
827       421         Lámpara de mesa  $114.83       9.0  05/04/2024   
828       424       Jarrón decorativo   $87.94       2.0  05/04/2024   
829      1868                Cafetera   $62.23       1.0  05/04/2024   
830      2545             Auriculares   $32.81      11.0  05/04/2024   
831      2778              SmartWatch   $70.66       7.0  05/04/2024   
844       145                Alfombra   $41.72       8.0  06/04/2024   
845       300         Adorno de pared   $67.75       9.0  06/04/2024   
846       439         Lámpara de mesa   $35.43       2.0  06/04/2024   
847       906                Cortinas   $84.04       3.0  06/04/2024   
848       959         Lámpara de mesa   $76.32       7.0  06/04/2024   
849      1376                Secadora   $69.58       7.0  06/04/2024   
850      1776                Secadora   $33.55       3.0  06/04/2024   
851      1784              Aspiradora  $115.24       3.0  06/04/2024   
852      1974              Aspiradora   $87.32       7.0  06/04/2024   
853      2080  Consola de videojuegos  $100.69       7.0  06/04/2024   
854      2108              SmartWatch   $79.37      10.0  06/04/2024   
855      2169               Proyector  $120.34       1.0  06/04/2024   
867       213       Espejo decorativo   $40.76       8.0  07/04/2024   
868       345                Alfombra   $56.78       9.0  07/04/2024   
869       398       Rincón de plantas   $77.30       7.0  07/04/2024   
870       509         Lámpara de mesa   $72.50       1.0  07/04/2024   
871       821         Lámpara de mesa   $71.17      10.0  07/04/2024   
872      1642                Heladera   $88.78       2.0  07/04/2024   
873      1956                Secadora   $42.59      12.0  07/04/2024   
874      2085             Auriculares  $122.56       6.0  07/04/2024   
875      2096                  Laptop   $83.25       6.0  07/04/2024   
876      2356                  Laptop   $30.16       6.0  07/04/2024   
877      2373              Smartphone  $123.60      12.0  07/04/2024   
884       204       Jarrón decorativo   $94.75       1.0  08/04/2024   
885      1381      Freidora eléctrica   $38.12       2.0  08/04/2024   
886      2365             Auriculares   $92.91      11.0  08/04/2024   
887      2486                  Laptop  $124.95      11.0  08/04/2024   
888      2506                  Laptop   $34.74       1.0  08/04/2024   
889      2705             Auriculares   $87.29       2.0  08/04/2024   

             categoria  
826         Decoración  
827         Decoración  
828         Decoración  
829  Electrodomésticos  
830        Electrónica  
831        Electrónica  
844         Decoración  
845         Decoración  
846         Decoración  
847         Decoración  
848         Decoración  
849  Electrodomésticos  
850  Electrodomésticos  
851  Electrodomésticos  
852  Electrodomésticos  
853        Electrónica  
854        Electrónica  
855        Electrónica  
867         Decoración  
868         Decoración  
869         Decoración  
870         Decoración  
871         Decoración  
872  Electrodomésticos  
873  Electrodomésticos  
874        Electrónica  
875        Electrónica  
876        Electrónica  
877        Electrónica  
884         Decoración  
885  Electrodomésticos  
886        Electrónica  
887        Electrónica  
888        Electrónica  
889        Electrónica

In [228]:
# Buscamos datos nulos
df_ventas.isnull().sum()

id_venta       0
producto       0
precio         2
cantidad       2
fecha_venta    0
categoria      0
dtype: int64

In [229]:
# Visualizar los nulos
# df_ventas[df_ventas.isnull().any(axis=1)]
df_ventas[df_ventas.isnull().any(axis=1)]

id_venta               producto precio  cantidad fecha_venta    categoria
136       627  Elementos de cerámica    NaN       NaN  17/01/2024   Decoración
139      2171    Parlantes Bluetooth    NaN       NaN  17/01/2024  Electrónica

In [230]:
# Analizar los precios de los NaN a ver que opciones tenemos
# df_ventas[df_ventas["producto"]=="Elementos de cerámica"]
df_ventas["producto"].value_counts()
# Unique nos devuelve una lista de los valores unicos.
# df_ventas["producto"].unique()


producto
Lámpara de mesa           181
Auriculares               147
Microondas                135
Cafetera                  118
Laptop                    104
Secadora                  103
Jarrón decorativo         102
Smartphone                102
Alfombra                  102
SmartWatch                102
Aspiradora                102
Cortinas                  102
Rincón de plantas         102
Freidora eléctrica        101
Espejo decorativo         101
Adorno de pared           101
Lavadora                  100
Consola de videojuegos    100
Cuadro decorativo         100
Horno eléctrico           100
Plancha de vapor          100
Proyector                 100
Elementos de cerámica     100
Batidora                  100
Televisor                 100
Cámara digital            100
Parlantes Bluetooth        99
Tablet                     58
Heladera                   49
Candelabro                 24
Name: count, dtype: int64

### Observaciones




* Cantidad de registros: 3035
* Columnas 6
* 35 duplicados, eliminarlos
* Algunos nulos (2), 1% del faltante, dropna
* Campo precio: encontre `$`, removerlo y luego cambiar el dtype a float64
* Campos cantidad: cambiar dtype a int64
* Campo fecha: ajustar a dtype datetime

## Dataframe marketing

### Analisis exploratorio de datos

Aplicamos los métodos de EDA vistos en clase, `info`, `columns`, `dtypes`, `head`, `tail`, `sample`, `unique`, `value_counts`, `describe`, `duplicated`, `isnull`, etc. y documentamos las observaciones:

In [231]:
# Aplicamos info() para familiarizarnos con el dataframe
df_marketing.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 90 entries, 0 to 89
Data columns (total 6 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   id_campanha   90 non-null     int64  
 1   producto      90 non-null     object 
 2   canal         90 non-null     object 
 3   costo         90 non-null     float64
 4   fecha_inicio  90 non-null     object 
 5   fecha_fin     90 non-null     object 
dtypes: float64(1), int64(1), object(4)
memory usage: 4.3+ KB


In [232]:
# Visualizamos los primeros registros con head()
# también podemos usar tail() para visualizar los últimos
df_marketing.head()

id_campanha         producto  canal  costo fecha_inicio   fecha_fin
0           74  Adorno de pared     TV   4.81   20/03/2024  03/05/2024
1           12           Tablet   RRSS   3.40   26/03/2024  13/05/2024
2           32  Lámpara de mesa  Email   5.54   28/03/2024  20/04/2024
3           21       Smartphone   RRSS   6.37   29/03/2024  16/05/2024
4           58         Alfombra  Email   4.25   31/03/2024  05/05/2024

In [233]:
# Analizamos el campo costo
df_marketing["costo"].sample(5)

45    4.88
25    4.52
8     5.08
63    5.00
16    3.96
Name: costo, dtype: float64

In [234]:
# Analizamos el campo fecha_inicio
df_marketing["fecha_inicio"].sample(5)

18    26/05/2024
23    31/05/2024
89    29/12/2024
45    03/08/2024
30    26/06/2024
Name: fecha_inicio, dtype: object

In [235]:
# Analizamos el campo fecha_fin
df_marketing["fecha_fin"].sample(5)

14    13/06/2024
15    29/05/2024
66    24/10/2024
62    02/11/2024
57    17/10/2024
Name: fecha_fin, dtype: object

### Calidad de los datos


In [236]:
# Buscamos registros duplicados
df_marketing.duplicated().sum()

np.int64(0)

### Observaciones
* Cantidad de registro: 90
* Columnas: 6
* Duplicado: 0
* Fecha inicio y fin: Cambiar a dtype datetime

## Dataframe clientes

In [237]:
# Aplicamos info()
df_clientes.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 567 entries, 0 to 566
Data columns (total 5 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   id_cliente  567 non-null    int64  
 1   nombre      567 non-null    object 
 2   edad        567 non-null    int64  
 3   ciudad      567 non-null    object 
 4   ingresos    567 non-null    float64
dtypes: float64(1), int64(2), object(2)
memory usage: 22.3+ KB


In [238]:
# Visualizamos con head()
df_clientes.head()

id_cliente               nombre  edad         ciudad  ingresos
0           1      Aloysia Screase    44  Mar del Plata  42294.68
1           2  Kristina Scaplehorn    25        Posadas  24735.04
2           3       Filip Castagne    50    Resistencia  35744.85
3           4          Liuka Luard    39   Bahía Blanca  27647.96
4           5        Dore Cockshtt    28        Rosario  28245.65

In [239]:
# Estadistica descriptiva
df_clientes.describe()

id_cliente        edad      ingresos
count  567.000000  567.000000    567.000000
mean   284.000000   37.940035  34668.739012
std    163.823075   10.202885  12974.531446
min      1.000000   20.000000    170.290000
25%    142.500000   30.000000  26015.240000
50%    284.000000   37.000000  35066.830000
75%    425.500000   43.000000  42457.100000
max    567.000000   81.000000  88053.010000

In [240]:
# Buscamos registros duplicados
df_clientes.duplicated().sum()

np.int64(0)

### Analisis exploratorio de datos

### Observaciones
* Cantidad de registros: 567
* Columnas: 5
* Duplicado: 0

# Etapa 2

## Limpieza de datos
Actividad 1 - Limpiar el conjunto de datos eliminando duplicados y caracteres no deseados. Documentar el proceso y los resultados.

### Dataframe ventas

* 35 duplicados, eliminarlos - se eliminaron 35 duplicados
* Algunos nulos (2), 1% del faltante, dropna - se eliminaron 2 registros con 2 NaN
* Cantidad de registros: 2998
* Campo precio: encontre `$`, removerlo y luego cambiar el dtype a float64 - Ok
* Campo cantidad: Cambiar el dtype a int64 - Ok
* Campo fecha: ajustar a dtype datetime - Ok

In [241]:
# Eliminamos registros duplicados
# atributos a evaluar, keep, ignore_index, inplace
df_ventas.drop_duplicates(inplace=True, ignore_index=True)

In [242]:
# Validamos que no haya duplicados
df_ventas.duplicated().sum()

np.int64(0)

In [243]:
# Eliminamos registros con datos nulos
df_ventas.dropna(inplace=True)

In [244]:
# Validamos que no haya nulos
# df_ventas.isnull().sum()
df_ventas.info()

<class 'pandas.core.frame.DataFrame'>
Index: 2998 entries, 0 to 2999
Data columns (total 6 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   id_venta     2998 non-null   int64  
 1   producto     2998 non-null   object 
 2   precio       2998 non-null   object 
 3   cantidad     2998 non-null   float64
 4   fecha_venta  2998 non-null   object 
 5   categoria    2998 non-null   object 
dtypes: float64(1), int64(1), object(4)
memory usage: 164.0+ KB


In [245]:
# Eliminamos el caracter especial $ de precio
# Normalización: usar replace()
# De ser necesario, podemos usar el atributo regex=True
df_ventas["precio"] = df_ventas["precio"].str.strip().str.replace("$", "")

In [246]:
# Validamos que precio no contenga $
df_ventas["precio"].sample(3)

80       32.34
666      33.31
2160    118.55
Name: precio, dtype: object

In [247]:
# Convertimos precio de str a float
# Podemos usar pd.to_numeric o astype o incluso combinarlos
df_ventas["precio"] = df_ventas["precio"].astype(float)

In [248]:
# Convertir el dtype de cantidad, de float64 a int64
df_ventas["cantidad"]= df_ventas["cantidad"].astype(int)

In [249]:
# Validar la conversion a int
df_ventas["cantidad"].dtypes

dtype('int64')

In [250]:
# Convertir el campo fecha_venta a dtype datetime
# pd.to_datetime(df_ventas["fecha_venta"], format="%d/%m/%Y")
# pd.to_datetime(df_ventas["fecha_venta"], dayfirst=True)
df_ventas['fecha_venta'] = pd.to_datetime(df_ventas['fecha_venta'], format="%d/%m/%Y")

In [251]:
# Validar la conversion de fecha
df_ventas['fecha_venta'].dtype

dtype('<M8[ns]')

### Dataframe marketing

* Cantidad de registro: 90
* Fecha inicio: Cambiar a dtype datetime - Ok
* Fecha fin: Cambiar a dtype datetime - Ok

In [252]:
# Convertir el campo fecha_inicio a dtype datetime
df_marketing['fecha_inicio'],pd.to_datetime(df_marketing['fecha_inicio'], format="%d/%m/%Y")

(0     20/03/2024
 1     26/03/2024
 2     28/03/2024
 3     29/03/2024
 4     31/03/2024
          ...    
 85    13/12/2024
 86    13/12/2024
 87    17/12/2024
 88    20/12/2024
 89    29/12/2024
 Name: fecha_inicio, Length: 90, dtype: object,
 0    2024-03-20
 1    2024-03-26
 2    2024-03-28
 3    2024-03-29
 4    2024-03-31
         ...    
 85   2024-12-13
 86   2024-12-13
 87   2024-12-17
 88   2024-12-20
 89   2024-12-29
 Name: fecha_inicio, Length: 90, dtype: datetime64[ns])

In [253]:
# Convertir el campo fecha_fin a dtype datetime
pd.to_datetime(df_marketing['fecha_fin'], format="%d/%m/%Y")

0    2024-05-03
1    2024-05-13
2    2024-04-20
3    2024-05-16
4    2024-05-05
        ...    
85   2024-12-29
86   2025-02-08
87   2025-02-14
88   2025-01-07
89   2025-01-21
Name: fecha_fin, Length: 90, dtype: datetime64[ns]

In [254]:
# Validamos los cambios
df_marketing['fecha_inicio'].dtype
df_marketing['fecha_fin'].dtype


dtype('O')

## Transformación de datos
Actividad 2 - Aplicar filtros y transformaciones para crear una tabla de ventas que muestre solo los productos con alto rendimiento.

In [255]:
df_ventas.head(4)

id_venta           producto  precio  cantidad fecha_venta  \
0       792  Cuadro decorativo   69.94         5  2024-01-02   
1       811    Lámpara de mesa  105.10         5  2024-01-02   
2      1156           Secadora   97.96         3  2024-01-02   
3      1372           Heladera  114.35         8  2024-01-02   

           categoria  
0         Decoración  
1         Decoración  
2  Electrodomésticos  
3  Electrodomésticos

In [256]:
# Transformación:
# Calculamos el valor de cada venta = precio * cantidad
# Agregamos una nueva columnas
df_ventas["valor_venta"] = df_ventas["precio"] * df_ventas["cantidad"]

In [257]:
# Visualizamos
df_ventas.head(2)

id_venta           producto  precio  cantidad fecha_venta   categoria  \
0       792  Cuadro decorativo   69.94         5  2024-01-02  Decoración   
1       811    Lámpara de mesa  105.10         5  2024-01-02  Decoración   

   valor_venta  
0        349.7  
1        525.5

In [258]:
# Agregación:
# Agrupar valor_venta por producto
# Sumamos las ventas dentro de cada grupo
tabla_ventas = (df_ventas.groupby("producto", as_index=False)["valor_venta"].sum())
tabla_ventas.head()

producto  valor_venta
0  Adorno de pared     48093.49
1         Alfombra     44773.06
2       Aspiradora     50085.86
3      Auriculares     74175.58
4         Batidora     50979.20

In [259]:
# Usemos describe para analizar algunos valores del dataframe resultante
tabla_ventas.describe()

valor_venta
count     30.000000
mean   48903.117333
std    13265.561715
min    11128.800000
25%    45601.920000
50%    48140.245000
75%    51092.960000
max    82276.380000

In [260]:
# Identificar productos que superan el umbral
# productos_top = tabla_ventas.query("valor_venta >= @umbral")
productos_top = tabla_ventas[tabla_ventas["valor_venta"] >= umbral_ventas]
productos_top

producto  valor_venta
3         Auriculares     74175.58
5            Cafetera     59607.31
9   Cuadro decorativo     54297.60
16  Jarrón decorativo     51130.88
19    Lámpara de mesa     82276.38
20         Microondas     72562.89
25           Secadora     52115.45
27         Smartphone     54132.44

## Agregación
Actividad 3 - Resumir las ventas por categoría de producto y analizar los ingresos generados.

In [261]:
df_ventas.columns

Index(['id_venta', 'producto', 'precio', 'cantidad', 'fecha_venta',
       'categoria', 'valor_venta'],
      dtype='object')

In [262]:
# Agrupamos por categoria y agregamos la suma
ventas_categoria = df_ventas.groupby("categoria", as_index=False)["valor_venta"].sum()
ventas_categoria.head()

categoria  valor_venta
0         Decoración    479216.09
1  Electrodomésticos    505299.63
2        Electrónica    482577.80

In [263]:
ventas_categoria

categoria  valor_venta
0         Decoración    479216.09
1  Electrodomésticos    505299.63
2        Electrónica    482577.80

In [264]:
# Ordenar y mostrar los resultados
ventas_categoria

categoria  valor_venta
0         Decoración    479216.09
1  Electrodomésticos    505299.63
2        Electrónica    482577.80

## Integración de datos
Actividad 4 - Combinar los sets de datos de ventas y marketing para obtener una visión más amplia de las tendencias.

In [265]:
df_ventas.head(2)

id_venta           producto  precio  cantidad fecha_venta   categoria  \
0       792  Cuadro decorativo   69.94         5  2024-01-02  Decoración   
1       811    Lámpara de mesa  105.10         5  2024-01-02  Decoración   

   valor_venta  
0        349.7  
1        525.5

In [266]:
df_marketing.head()

id_campanha         producto  canal  costo fecha_inicio   fecha_fin
0           74  Adorno de pared     TV   4.81   20/03/2024  03/05/2024
1           12           Tablet   RRSS   3.40   26/03/2024  13/05/2024
2           32  Lámpara de mesa  Email   5.54   28/03/2024  20/04/2024
3           21       Smartphone   RRSS   6.37   29/03/2024  16/05/2024
4           58         Alfombra  Email   4.25   31/03/2024  05/05/2024

In [267]:
# Unir los datasets ventas y marketing por columna 'producto'
ventas_marketing = pd.merge(df_ventas, df_marketing, on="producto", how="left")

In [268]:
# Verificar coincidencias (muestra)
ventas_marketing.head()

id_venta           producto  precio  cantidad fecha_venta   categoria  \
0       792  Cuadro decorativo   69.94         5  2024-01-02  Decoración   
1       792  Cuadro decorativo   69.94         5  2024-01-02  Decoración   
2       792  Cuadro decorativo   69.94         5  2024-01-02  Decoración   
3       811    Lámpara de mesa  105.10         5  2024-01-02  Decoración   
4       811    Lámpara de mesa  105.10         5  2024-01-02  Decoración   

   valor_venta  id_campanha  canal  costo fecha_inicio   fecha_fin  
0        349.7            1   RRSS   5.27   27/04/2024  04/06/2024  
1        349.7           31  Email   5.28   15/08/2024  12/09/2024  
2        349.7           61     TV   5.30   05/11/2024  23/12/2024  
3        525.5           32  Email   5.54   28/03/2024  20/04/2024  
4        525.5            2   RRSS   5.88   30/05/2024  29/06/2024

### Analizamos el impacto de la campaña

In [275]:
# Nos quedamos solo con los registros de ventas durante la campaña de marketing
# Seleccionamos solo los campos que nos interesan: "producto", "valor_venta", "costo"
df_filtro_marketing = ventas_marketing[ventas_marketing["id_campanha"].between(1, 90)].copy()


In [276]:
# Agrupamos por producto y hacemos una agregación por "valor de venta" y "costo"
ventas_marketing.groupby("categoria")[["valor_venta","costo"]].sum()

valor_venta     costo
categoria                               
Decoración          1437648.27  15774.86
Electrodomésticos   1515898.89  14644.65
Electrónica         1447733.40  14243.90

In [283]:
# Generamos una nueva columna con la diferencia entre lo vendido y lo invertido en la campaña
df_filtro_marketing["ganancia"] = df_filtro_marketing["valor_venta"] - df_filtro_marketing["costo"]
# Verificamos
df_filtro_marketing.head(10)

id_venta           producto  precio  cantidad fecha_venta  \
0       792  Cuadro decorativo   69.94         5  2024-01-02   
1       792  Cuadro decorativo   69.94         5  2024-01-02   
2       792  Cuadro decorativo   69.94         5  2024-01-02   
3       811    Lámpara de mesa  105.10         5  2024-01-02   
4       811    Lámpara de mesa  105.10         5  2024-01-02   
5       811    Lámpara de mesa  105.10         5  2024-01-02   
6      1156           Secadora   97.96         3  2024-01-02   
7      1156           Secadora   97.96         3  2024-01-02   
8      1156           Secadora   97.96         3  2024-01-02   
9      1372           Heladera  114.35         8  2024-01-02   

           categoria  valor_venta  id_campanha  canal  costo fecha_inicio  \
0         Decoración       349.70            1   RRSS   5.27   27/04/2024   
1         Decoración       349.70           31  Email   5.28   15/08/2024   
2         Decoración       349.70           61     TV   5.30   05/11/2024   
3         Decoración       525.50           32  Email   5.54   28/03/2024   
4         Decoración       525.50            2   RRSS   5.88   30/05/2024   
5         Decoración       525.50           62     TV   4.51   26/08/2024   
6  Electrodomésticos       293.88            3   RRSS   5.10   24/07/2024   
7  Electrodomésticos       293.88           63     TV   5.69   03/11/2024   
8  Electrodomésticos       293.88           33  Email   3.80   20/12/2024   
9  Electrodomésticos       914.80           34  Email   4.20   06/05/2024   

    fecha_fin  ganancia  
0  04/06/2024    344.43  
1  12/09/2024    344.42  
2  23/12/2024    344.40  
3  20/04/2024    519.96  
4  29/06/2024    519.62  
5  07/10/2024    520.99  
6  30/08/2024    288.78  
7  28/11/2024    288.19  
8    7/1/2025    290.08  
9  05/07/2024    910.60